apps/frontend/src/components/questionary/questionaryComponents/InstrumentPicker/QuestionaryComponentInstrumentPicker.tsx

```diff
@@ -62,9 +116,137 @@ export function QuestionaryComponentInstrumentPicker(
   );
 
   const handleOnChange = (event: SelectChangeEvent<string | string[]>) => {
-    const newValue = event.target.value || null;
+    const newValue = event.target.value;
+    let newInstrumentTime: InstrumentIdAndTime[] | InstrumentIdAndTime =
+      [] || {};
+    if (
+      Array.isArray(newValue) &&
+      newValue.length > 0 &&
+      Array.isArray(requestTimeForInstrument)
+    ) {
+      newInstrumentTime = newValue.map((id) => {
+        return {
+          instrumentId: id,
+          timeRequested: requestTimeForInstrument.filter(
+            (value) => value.instrumentId === id
+          )
+            ? requestTimeForInstrument.find(
+                (value) => value.instrumentId === id
+              )?.timeRequested
+            : '0',
+        };
+      });
+    } else if (typeof newValue === 'string') {
+      newInstrumentTime = {
+        instrumentId: newValue,
+        timeRequested: '0',
+      };
+    }
+    onComplete(newInstrumentTime);
+  };
 
-    onComplete(newValue);
+  const handleTimeInput = (time: string, id: string) => {
+    let newInstrumentTime: InstrumentIdAndTime[] | InstrumentIdAndTime =
+      [] || {};
+    newInstrumentTime = Array.isArray(requestTimeForInstrument)
+      ? requestTimeForInstrument.map((obj) => {
+          if (obj.instrumentId === id) {
+            return { instrumentId: obj.instrumentId, timeRequested: time };
+          } else {
+            return {
+              instrumentId: obj.instrumentId,
+              timeRequested: obj.timeRequested,
+            };
+          }
+        })
+      : { instrumentId: id, timeRequested: time };
+    onComplete(newInstrumentTime);
+  };
+  const DynamicTimeFields = () => {
+    return requestTimeForInstrument &&
+      Array.isArray(requestTimeForInstrument) ? (
+      <Stack direction="row" spacing={3} marginTop={3}>
+        {requestTimeForInstrument.map((value) => {
+          if (value.instrumentId)
+            return (
+              <TextField
+                key={value.instrumentId}
+                value={value.timeRequested === '0' ? '' : value.timeRequested}
+                required={config.required}
+                error={isError}
+                label={`Request Time`}
+                type="number"
+                InputProps={{
+                  startAdornment: (
+                    <InputAdornment position="start">
+                      {value.instrumentName}:
+                    </InputAdornment>
+                  ),
+                }}
+                data-time-request={value.instrumentId + '-time-request'}
+                onChange={(e: React.ChangeEvent<HTMLTextAreaElement>) => {
+                  handleTimeInput(e.target.value, value.instrumentId);
+                }}
+              />
+            );
+        })}
+      </Stack>
+    ) : (
+      <Stack direction="row" spacing={3} marginTop={3}>
```

Comment:
Instead of duplicating this code as it looks quite similar with the one where we map the `requestTimeForInstrument` can't we use map in both cases and just make the `requestTimeForInstrument` as an array if it is not. Let me know if this is a bit confusing to understand I will try to explain better.